packages/two_dimensional_scrollables/lib/src/tree_view/tree.dart

```diff
@@ -0,0 +1,1056 @@
+  /// is used.
+  final TreeViewRowBuilder treeRowBuilder;
+
+  /// If provided, the controller can be used to expand and collapse
+  /// [TreeViewNode]s, or lookup information about the current state of the
+  /// [TreeView].
+  final TreeViewController? controller;
+
+  /// Called when a [TreeViewNode] expands or collapses.
+  ///
+  /// This will not be called if a [TreeViewNode] does not have any children.
+  final TreeViewNodeCallback? onNodeToggle;
+
+  /// Whether to wrap each row of the tree in an [AutomaticKeepAlive].
+  ///
+  /// Typically, lazily laid out children are wrapped in [AutomaticKeepAlive]
+  /// widgets so that the children can use [KeepAliveNotification]s to preserve
+  /// their state when they would otherwise be garbage collected off-screen.
+  ///
+  /// This feature (and [addRepaintBoundaries]) must be disabled if the children
+  /// are going to manually maintain their [KeepAlive] state. It may also be
+  /// more efficient to disable this feature if it is known ahead of time that
+  /// none of the children will ever try to keep themselves alive.
+  ///
+  /// Defaults to true.
+  final bool addAutomaticKeepAlives;
+
+  /// Whether to wrap each row in a [RepaintBoundary].
+  ///
+  /// Typically, children in a scrolling container are wrapped in repaint
+  /// boundaries so that they do not need to be repainted as the list scrolls.
+  /// If the children are easy to repaint (e.g., solid color blocks or a short
+  /// snippet of text), it might be more efficient to not add a repaint boundary
+  /// and instead always repaint the children during scrolling.
+  ///
+  /// Defaults to true.
+  final bool addRepaintBoundaries;
+
+  /// Used to override the toggle animation's curve and duration.
+  ///
+  /// If [AnimationStyle.duration] is provided, it will be used to override
+  /// the [TreeView.defaultAnimationDuration], which is 150
+  /// milliseconds.
+  ///
+  /// If [AnimationStyle.curve] is provided, it will be used to override
+  /// the [TreeView.defaultAnimationCurve], which is [Curves.linear].
+  ///
+  /// To disable the tree animation, use [AnimationStyle.noAnimation].
+  final AnimationStyle? animationStyle;
+
+  /// A default of [Curves.linear], which is used in the tree's expanding and
+  /// collapsing node animation.
+  static const Curve defaultAnimationCurve = Curves.linear;
+
+  /// A default [Duration] of 150 milliseconds, which is used in the tree's
+  /// expanding and collapsing node animation.
+  static const Duration defaultAnimationDuration = Duration(milliseconds: 150);
+
+  /// The order in which [TreeViewNode]s are visited.
+  ///
+  /// This value will influence [TreeViewport.mainAxis] so nodes are traversed
+  /// properly when they are converted to a [TreeVicinity] in the context of the
+  /// active nodes of the tree.
+  ///
+  /// Defaults to [TreeViewTraversalOrder.depthFirst].
+  final TreeViewTraversalOrder traversalOrder;
```

Comment:
I am not sure how else to describe it. In TableView, mainAxis determines what order we visit children in row major or column major order, visit all the children along a row first, or visit all the children along a column fist.

Here, we don't have rows and columns, just rows, but varying depths. So the ChildVicinity (TreeVicinity) reflects the row index and the depth. Translated into row major or column major order, this results in breadth or depth first traversal order.

Reply:
Per an offline discussion, we agreed to remove the traversal property/class since there may not be a lot of demand for this feature. I'll file an issue to follow up and see if there is demand for it. 👍